vision_datasets/image_object_detection/detection_as_kvp_dataset.py

```diff
@@ -44,24 +61,51 @@ def __init__(self, detection_dataset: VisionDataset):
 
             kvp_label_data = self.construct_kvp_label_data(bboxes)
             img_ids = [self.img_id_to_pos[img.id]]  # 0-based index
-            kvp_annotation = KeyValuePairLabelManifest(id, img_ids, label_data=kvp_label_data)
+            kvp_annotation = KeyValuePairLabelManifest(
+                id, img_ids, label_data=kvp_label_data
+            )
 
             # KVPDatasetManifest expects img.labels to be empty. Labels are instead stored in KVP annotation
             img.labels = []
             annotations.append(kvp_annotation)
 
-        dataset_manifest = KeyValuePairDatasetManifest(detection_dataset.dataset_manifest.images, annotations, schema, additional_info=detection_dataset.dataset_manifest.additional_info)
-        super().__init__(dataset_info, dataset_manifest, dataset_resources=detection_dataset.dataset_resources)
-
-    def construct_schema(self, class_names: List[str]) -> Dict[str, Any]:
+        dataset_manifest = KeyValuePairDatasetManifest(
+            detection_dataset.dataset_manifest.images,
+            annotations,
+            schema,
+            additional_info=detection_dataset.dataset_manifest.additional_info,
+        )
+        super().__init__(
+            dataset_info,
+            dataset_manifest,
+            dataset_resources=detection_dataset.dataset_resources,
+        )
+
+    def construct_schema(
+        self, include_class_names: bool, custom_schema_description: Union[str, None]
```

Comment:
I noticed that `class_names` is being removed from the argument of `construct_schema`. Is it intentional? I suppose this method could be a `@classmethod` if it is on the argument?

Reply:
Since we are anyway constructing `self.class_names` in the initializer, I could directly reference it inside the construct_schema method. No strong opinion though.